store/ssmstore.go

```diff
@@ -126,6 +127,72 @@ func (s *SSMStore) Read(ctx context.Context, id SecretId, version int) (Secret,
 	return s.readVersion(ctx, id, version)
 }
 
+func (s *SSMStore) WriteTags(ctx context.Context, id SecretId, tags map[string]string, deleteOtherTags bool) error {
+	// list the current tags
+	currentTags, err := s.ReadTags(ctx, id)
```

Comment:
we only use the current tags if we're doing an idempotent tag update and deleting other tags, right?

If so, can we only ready the tags when we're going to do something with those current tags?

Reply:
Good catch! Fixed.